src/Examples/Flowsheets/HDA_flowsheet_with_costing.ipynb

```diff
@@ -0,0 +1,1598 @@
+   "source": [
+    "## Add Operating Cost Equations\n",
+    "Before adding capital costing blocks, we will add operating cost equations taken from the prior examples. The examples assume constant cooling and heating coefficients over an annual cost basis. The IDAES Generic Costing Framework does not currently support variable cost calculations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# required imports\n",
+    "from pyomo.environ import Expression\n",
+    "\n",
+    "# operating costs for HDA with second flash (model m)\n",
+    "m.fs.cooling_cost = Expression(expr=0.212e-7 * (-m.fs.F101.heat_duty[0]) +\n",
+    "                                   0.212e-7 * (-m.fs.R101.heat_duty[0]))\n",
+    "m.fs.heating_cost = Expression(expr=2.2e-7 * m.fs.H101.heat_duty[0] +\n",
+    "                                   1.9e-7 * m.fs.F102.heat_duty[0])\n",
+    "m.fs.operating_cost = Expression(expr=(3600 * 24 * 365 *\n",
+    "                                           (m.fs.heating_cost +\n",
+    "                                            m.fs.cooling_cost)))\n",
+    "\n",
+    "# operating costs for HDA with distillation (model n)\n",
+    "n.fs.cooling_cost = Expression(expr=0.25e-7 * (-n.fs.F101.heat_duty[0]) +\n",
+    "                                0.2e-7 * (-n.fs.D101.condenser.heat_duty[0]))\n",
+    "\n",
+    "n.fs.heating_cost = Expression(expr=2.2e-7 * n.fs.H101.heat_duty[0] +\n",
+    "                                1.2e-7 * n.fs.H102.heat_duty[0] +\n",
+    "                                1.9e-7 * n.fs.D101.reboiler.heat_duty[0])\n",
+    "\n",
+    "n.fs.operating_cost = Expression(expr=(3600 * 24 * 365 *\n",
+    "                                (n.fs.heating_cost + n.fs.cooling_cost)))   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Add Capital Costing\n",
+    "Below, we will add add rigorous capital costing blocks to the imported flowsheets and evaluate the economic impact of replacing the second Flash with a Distillation column. First, let's import the relevant costing methods:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import costing methods - classes, heaters, vessels, compressors, columns\n",
+    "from idaes.models.costing.SSLW import (\n",
+    "    SSLWCosting,\n",
+    "    SSLWCostingData,\n",
+    "    VesselMaterial,\n",
+    "    TrayType,\n",
+    "    TrayMaterial,\n",
+    "    HeaterMaterial,\n",
+    "    HeaterSource,\n",
+    "    CompressorDriveType,\n",
+    "    CompressorMaterial,\n",
+    "    CompressorType,\n",
+    ")\n",
+    "from idaes.core import UnitModelCostingBlock"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we will build the main costing block for each flowsheet, calling the main costing class to define the blocks:"
```

Comment:
You should briefly explain what this block will do (hold global parameters, keep track of costed equipment, aggregate unit costs, etc.)